service/history/execution/mutable_state_util_test.go

```diff
@@ -24,13 +24,100 @@ import (
 	"testing"
 	"time"
 
+	fuzz "github.com/google/gofuzz"
 	"github.com/stretchr/testify/assert"
 
 	"github.com/uber/cadence/common"
 	"github.com/uber/cadence/common/clock"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/testing/testdatagen"
 	"github.com/uber/cadence/common/types"
 )
 
+func TestCopyActivityInfo(t *testing.T) {
+	for i := 0; i < 100; i++ {
+		t.Run("test CopyActivityInfo Mapping", func(t *testing.T) {
+			f := fuzz.New().Funcs(testdatagen.GenHistoryEvent).NilChance(0.2)
+
+			d1 := persistence.ActivityInfo{}
+			f.Fuzz(&d1)
+			d2 := CopyActivityInfo(&d1)
+
+			assert.Equal(t, &d1, d2)
+		})
+	}
+}
+
+func TestCopyWorkflowExecutionInfo(t *testing.T) {
+	for i := 0; i < 100; i++ {
```

Comment:
hmmm.  obviously this works (hopefully does different values each time too?), but I'm not sure we gain anything from N tests vs N loops in one test... neither is easy to short-circuit or narrow down when reproducing something.